### DISCUSSION TITLE
How to send data from middleware to filter?

### DISCUSSION THREAD

**Discussion by Dugers** [aiogram/aiogram#957] [Q&A]
My middleware:
```python
class UserRoleMiddleware(BaseMiddleware):
    async def on_pre_process_message(self, message: Message, data):
        data['user_info'] = query to db
```

**Comment by merura** [comment]
[Source (ru)](https://te.legra.ph/aiogram-ru-01-28#%D0%9A%D0%B0%D0%BA-%D0%BF%D0%BE%D0%BB%D1%83%D1%87%D0%B8%D1%82%D1%8C-%D0%B4%D0%B0%D0%BD%D0%BD%D1%8B%D0%B5-%D0%B8%D0%B7-%D0%BC%D0%B8%D0%B4%D0%BB%D0%B2%D0%B0%D1%80%D0%B8-%D0%B2-%D1%84%D0%B8%D0%BB%D1%8C%D1%82%D1%80%D0%B0%D1%85?)
```python
from aiogram.dispatcher.handler import ctx_data
...
# in your filter
data = ctx_data.get()
data["user_info"]
```
It is only for aiogram 2.x

**Comment by DanilVinsmok** [comment]
How can I access data from middleware in aiogram 3.x?

**Comment by JrooTJunior** [comment]
@DanilVinsmok, https://docs.aiogram.dev/en/dev-3.x/dispatcher/dependency_injection.html
